Samples/ShareTarget/cs/ShareTarget.xaml.cs

```diff
@@ -286,6 +286,25 @@ public ShareTarget()
                         {
                             AddContentValue("HTML: ");
                             ShareWebView.Visibility = Visibility.Visible;
+                            // Check if there are any local images in the resource map.
+                            if (this.sharedResourceMap.Count > 0)
+                            {
+                                foreach (KeyValuePair<string, RandomAccessStreamReference> item in this.sharedResourceMap)
+                                {
+                                    var stream = await item.Value.OpenReadAsync();
+                                    if (!stream.ContentType.Contains("image/"))
+                                        continue;
+                                    var reader = new DataReader(stream.GetInputStreamAt(0));
+                                    await reader.LoadAsync((uint)stream.Size);
+                                    byte[] byteArray = new byte[stream.Size];
+                                    reader.ReadBytes(byteArray);
+                                    String base64String = "<img src ='data:image/gif;base64," + Convert.ToBase64String(byteArray) + "'";
+                                    String replaceTarget = "<img src=" + item.Key;
+
+                                    htmlFragment = htmlFragment.Replace(replaceTarget, base64String);
```

Comment:
Would it be better to use a stream resolver? Then you won't run into trouble if the image is large and exceeds the maximum length of a data: URI.